crates/e2e/macro/src/config.rs

```diff
@@ -128,18 +136,59 @@ mod tests {
     }
 
     #[test]
-    fn duplicate_args_fails() {
+    fn duplicate_additional_contracts_fails() {
         assert_try_from(
             syn::parse_quote! {
                 additional_contracts = "adder/Cargo.toml",
                 additional_contracts = "adder/Cargo.toml",
             },
             Err(
-                "encountered duplicate ink! e2e test `additional_contracts` configuration argument",
+                "encountered duplicate ink! E2E test `additional_contracts` configuration argument",
+            ),
+        );
+    }
+
+    #[test]
+    fn duplicate_environment_fails() {
+        assert_try_from(
+            syn::parse_quote! {
+                environment = crate::CustomEnvironment,
+                environment = crate::CustomEnvironment,
+            },
+            Err(
+                "encountered duplicate ink! E2E test `environment` configuration argument",
             ),
         );
     }
 
+    #[test]
+    fn environment_as_literal_fails() {
+        assert_try_from(
+            syn::parse_quote! {
+                environment = "crate::CustomEnvironment",
+            },
+            Err("expected a path for `environment` ink! E2E test configuration argument"),
+        );
+    }
+
+    #[test]
+    fn full_config_works() {
+        assert_try_from(
+            syn::parse_quote! {
+                additional_contracts = "adder/Cargo.toml flipper/Cargo.toml",
+                environment = crate::CustomEnvironment,
```

Comment:
So the fact that this isn't a string literal makes it a bit inconsistent imo. 

At the same time we're still early with the E2E framework and can adjust the parameters in the future as it gets more usage and eyeballs

Reply:
while a path to manifest cannot be well expressed in a type from `syn`, a path can be, so it's more natural for me to use semantic type where it's possible

Reply:
Yeah I know, but it's unfortunate that users will have to remember things they should and shouldn't be putting into quotes, as opposed to just having one "style" throughout. Not saying you need to change your implementation or anything, just thinking out loud about future improvements